File: app/Models/Participation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Participation extends Model
{
    use HasFactory;

    protected $fillable = [
        'student_id',
        'event_id',
        'time_in',
        'time_out'
    ];
    protected $casts = [
        'time_in' => 'datetime',
        'time_out' => 'datetime',
    ];

    public function student()
    {
        return $this->belongsTo(Student::class);
    }

    public function event()
    {
        return $this->belongsTo(Event::class);
    }
}

File: mobile_flutter/lib/models/participation.dart
class Participation {
  final int id;
  final int studentId;
  final int eventId;
  final String timeIn;
  final String? timeOut;

  Participation({
    required this.id,
    required this.studentId,
    required this.eventId,
    required this.timeIn,
    this.timeOut,
  });

  factory Participation.fromJson(Map<String, dynamic> json) {
    return Participation(
      id: json['id'],
      studentId: json['student_id'],
      eventId: json['event_id'],
      timeIn: json['time_in'],
      timeOut: json['time_out'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'student_id': studentId,
      'event_id': eventId,
      'time_in': timeIn,
      'time_out': timeOut,
    };
  }
}
